my-react-app/src/entry-client.tsx

```diff
@@ -0,0 +1,19 @@
+import 'normalize.css';
+import './index.css';
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import App from './App';
+import { BrowserRouter } from 'react-router-dom';
+import { store } from './store/store';
+import { Provider } from 'react-redux';
+
+ReactDOM.hydrateRoot(
+  document.getElementById('root') as HTMLElement,
+  <React.StrictMode>
+    <BrowserRouter>
+      <Provider store={store}>
```

Comment:
you did not adopt Redux for SSR.

https://redux.js.org/usage/server-rendering#inject-initial-component-html-and-state
https://react-redux.js.org/api/provider#react-18-ssr-hydration